### DISCUSSION TITLE
some issues with starting programs on the AsusWRT-Merlin.

### DISCUSSION THREAD

**Discussion by dasfdsa123** [v2rayA/v2rayA#1487] [Q&A]
version : v2raya - 2.2.5.8-1 

I installed  v2raya ,v2ray-core, v2ray-geoip, and v2ray-geosite using opkg.
However, the program cannot start properly.

err log:
2024/08/27 04:16:44.484  [E] [once.go:74]  bbolt.Open: invalid argument
2024/08/27 04:17:59.346  [E] [once.go:74]  bbolt.Open: invalid argument
2024/08/27 04:27:21.931  [E] [once.go:74]  bbolt.Open: invalid argument
2024/08/27 04:39:38.203  [E] [once.go:74]  bbolt.Open: invalid argument

```
init.d:
#!/bin/sh

# V2Ray binary is xray by default (if xray installed)
if [ -x "/opt/sbin/xray" ]; then
  V2RAY_BIN="-b /opt/sbin/xray"
elif [ -x "/opt/sbin/v2ray" ]; then
  V2RAY_BIN="-b /opt/sbin/v2ray"
else
  echo "v2ray or xray not installed" && exit
fi

ENABLED=yes
PROCS=v2raya
ARGS="$V2RAY_BIN -c /opt/etc/v2raya --v2ray-assetsdir /opt/share/v2ray
--nftables-support off --log-file /opt/var/log/v2raya.log"
PREARGS=""
DESC=$PROCS
PATH=/opt/sbin:/opt/bin:/opt/usr/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin

. /opt/etc/init.d/rc.func


```

**Comment by dushez** [comment]
在boltdb.go文件找到这段代码
`
func initDB() {
	confPath := conf.GetEnvironmentConfig().Config
	dbPath := filepath.Join(confPath, "bolt.db")
	if readOnly {
		// trick: not really read-only
		f, err := os.CreateTemp(os.TempDir(), "v2raya_tmp_bolt_*.db")
		if err != nil {
			panic(err)
		}
		newPath := f.Name()
		f.Close()
		if err = copyfile.CopyFileContent(dbPath, newPath); err != nil {
			panic(err)
		}
		dbPath = newPath
	}

	var err error
	db, err = bbolt.Open(dbPath, 0600, nil)
	if err != nil {
		log.Fatal("bbolt.Open: %v", err)
	}
}
`

打开的文件是bolt.db
在路由 /etc/v2raya 目录下文件名是boltv4.db
将文件名改为bolt.db后在web页v2raya管理页面将服务重启
